### PR TITLE
change arguments order

### PR DESCRIPTION
**Pull request by fanatid**
Just my vision of this package. #1 #3

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,49 +1,53 @@
...
-  // compression flag?
-  if (buffer.length === 34) {
-    if (buffer[33] !== 0x01) throw new Error('Invalid compression flag')
+function decodeRaw (buffer, version) {
+  // check version only if defined
+  if (version !== undefined && buffer[0] !== version) {
+    throw new Error('Invalid network version')
+  }
 
-    // truncate the version byte/compression flag
```
Comment (dcousens) [inline index.js]: why was all this subsequent code changed OOI?

File: index.js [hunk 2]
```diff
@@ -1,49 +1,53 @@
...
   }
 
-  return buffer
+  return result
 }
 
-function encode (version, d, compressed) {
-  return bs58check.encode(encodeRaw(version, d, compressed))
+function encode (version, buffer, compressed) {
+  if (Object.prototype.toString.call(version) === '[object Object]') {
```
Comment (dcousens) [inline index.js]: `typeof version !== 'number'`?

File: index.js (hunk 1 above)
Comment (fanatid) [inline index.js]: New code is more cleaner, isn't it?
I change `d` to `buffer` because if it was `d` it should be a `Integer`

Comment (dcousens) [inline index.js]: True,  I'll review this more closely later today :+1:

File: index.js [hunk 3]
```diff
@@ -1,49 +1,53 @@
...
+    throw new Error('Invalid network version')
+  }
 
-    // truncate the version byte/compression flag
+  if (buffer.length === 33 || (buffer.length === 34 && buffer[33] === 0x01)) {
     return {
       version: buffer[0],
-      d: buffer.slice(1, -1),
-      compressed: true
+      buffer: buffer.slice(1, 33),
```
Comment (dcousens) [inline index.js]: @jprichardson happy with this?

Comment (jprichardson) [inline index.js]: I think `buffer` lacks clarity on what. Yes, it gives a strong indication on the type, but I think that's less important as to what it is. Why not `privateKey`?

Comment (fanatid) [inline index.js]: I don't mind rename `buffer` to `privateKey`

**Comment by fanatid** [comment]
Updated. Now this PR solve all current open issues  :smiley:

**Comment by dcousens** [comment]
Published as `2.0.0`

**Pull request merged**